receiver/awsfirehosereceiver/internal/unmarshaler/cwmetricstream/unmarshaler.go

```diff
@@ -113,6 +114,33 @@ func (u Unmarshaler) UnmarshalMetrics(record []byte) (pmetric.Metrics, error) {
 		maxQ := dp.QuantileValues().AppendEmpty()
 		maxQ.SetQuantile(1)
 		maxQ.SetValue(cwMetric.Value.Max)
+
+		for key, value := range cwMetric.Value.Percentiles {
+			// Only process percentile fields (those starting with 'p')
+			if len(key) < 3 || key[0] != 'p' {
```

Comment:
```suggestion
			if len(key) < 2 || key[0] != 'p' {
```

I suppose fairly unlikely, but "p5" (say) would fail otherwise.